provider/src/main/java/com/raygun/raygun4android/network/RaygunNetworkUtils.kt

```diff
@@ -0,0 +1,88 @@
+package com.raygun.raygun4android.network
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.net.ConnectivityManager
+import android.provider.Settings
+import androidx.annotation.VisibleForTesting
+import androidx.core.content.edit
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.net.HttpURLConnection
+import java.net.URLConnection
+import java.nio.charset.StandardCharsets
+import java.util.UUID
+
+internal interface UuidProvider {
+    suspend fun getDeviceUuid(context: Context): String
+}
+
+object RaygunNetworkUtils {
+    private const val PREFS_FILE = "device_id.xml"
+    private const val PREFS_DEVICE_ID = "device_id"
+
+    fun getStatusCode(urlConnection: URLConnection?): Int {
+        var statusCode = 0
+        if (urlConnection != null) {
+            if ((urlConnection is HttpURLConnection)) {
+                try {
+                    statusCode = urlConnection.responseCode
+                } catch (ignore: Exception) {
+                }
+            }
+        }
+        return statusCode
+    }
+
+    @JvmStatic
+    fun hasInternetConnection(appContext: Context): Boolean {
+        val cm =
+            appContext.applicationContext.getSystemService(Context.CONNECTIVITY_SERVICE)
+                as ConnectivityManager?
+
+        if (cm != null) {
+            val activeNetwork = cm.activeNetworkInfo
+            return activeNetwork != null && activeNetwork.isConnected
+        }
+
+        return false
+    }
+
+    suspend fun getDeviceUuid(context: Context): String = uuidProvider.getDeviceUuid(context)
+
+    @VisibleForTesting
+    internal var uuidProvider: UuidProvider =
```

Comment:
This allows to override the `uuidProvider` for tests, facilitates mocking, since mocking static suspend methods gets complex